### PR TITLE
[FLINK-26152] [docs] Translate the page of SQL/queries/WITH clause

### PR DESCRIPTION
**Pull request by magicWyb**
## What is the purpose of the change

*(For example: This pull request makes task deployment go through the blob server, rather than through RPC. That way we avoid re-transferring them on each deployment (during recovery).)*

## Brief change log

*(for example:)*
  - *The TaskInfo is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *TaskManagers retrieve the TaskInfo from the blob cache*

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluser with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: docs/content.zh/docs/dev/table/sql/queries/with.md [hunk 1]
```diff
@@ -25,9 +25,9 @@ under the License.
 # WITH 语句
 {{< label Batch >}} {{< label Streaming >}}
 
-`WITH` provides a way to write auxiliary statements for use in a larger query. These statements, which are often referred to as Common Table Expression (CTE), can be thought of as defining temporary views that exist just for one query.
+`WITH` 语句提供了一种用于更大查询而编写辅助语句的方法。这些编写的语句通常被称为公用表表达式，表达式可以理解为仅针对某个查询而存在的临时视图。
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/table/sql/queries/with.md:28]:
```
`WITH` 语句提供了一种编写辅助语句用于更大查询的方式。这些编写的语句通常被称为公用表表达式（Common Table Expression (CTE)），表达式可以理解为仅针对某个查询而存在的临时视图。
```

File: docs/content.zh/docs/dev/table/sql/queries/with.md [hunk 2]
```diff
@@ -25,9 +25,9 @@ under the License.
 # WITH 语句
```
Comment (RocMarshal) [inline docs/content.zh/docs/dev/table/sql/queries/with.md:25]:
```suggestion
<a name="with-clause"></a>

# WITH 语句
```

And what do you think about translating 'clause' into '子句' ?

**Review by RocMarshal — COMMENTED** [review]
@magicWyb Thanks for the contribution. I left some comments, please let know what's your opinion.

**Comment by magicWyb** [comment]
I modified the translation of WITH, previously translated as WITH 语句，now translated as WITH 子句。

**Review by RocMarshal — COMMENTED** [review]
@magicWyb Thanks for the update.
And would you mind checking the CI report result.
Maybe we should activate the CI to re-run by `@flinkbot run azure`

**Comment by magicWyb** [comment]
I don't mind checking the CI report result. But I don't know how to do it, because this is my first commit pull request.
And i would like to learn how to do it.

**Comment by RocMarshal** [comment]
@flinkbot run azure

**Comment by magicWyb** [comment]
I see it in this url  https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=31915&view=results.
And  almost 95% test complete.

**Review by RocMarshal — APPROVED** [review]
@magicWyb LGTM (non-binding)

@gaoyunhaii @wuchong . 
Could you help to make the final check ?

**Comment by wuchong** [comment]
Thanks @RocMarshal and @magicWyb . I will merge it.

**Pull request merged**